fastcrud/endpoint/helper.py

```diff
@@ -43,10 +43,24 @@ def check_valid_method(cls, values: Sequence[str]) -> Sequence[str]:
 
 
 def _get_primary_key(model: type[DeclarativeBase]) -> Union[str, None]:
+    return _get_primary_keys(model)[0].name
+
+
+def _get_primary_keys(model: type[DeclarativeBase]) -> Sequence[Column]:
     """Get the primary key of a SQLAlchemy model."""
     inspector = inspect(model).mapper
     primary_key_columns = inspector.primary_key
-    return primary_key_columns[0].name if primary_key_columns else None
+
+    # Some patching when using derived sa.TypeDecorator as primary keys
```

Comment:
Maybe we should go for a safer (and more explicit) approach here, something like:

```python
for pk in primary_key_columns:
        try:
            _ = pk.type.python_type
        
        except NotImplementedError:
            if hasattr(pk.type, 'impl') and hasattr(pk.type.impl, 'python_type'):
                logging.warning(f"Missing python_type for primary key {pk.name}. Attempting to use `impl.python_type` as a fallback.")
            else:
                raise NotImplementedError(f"The primary key column {pk.name} uses a custom type without a defined `python_type` or suitable `impl` fallback.")

    return primary_key_columns
```

It's probably even better to just move this to a separate `_get_python_type` function or something like it:

```python
def _get_python_type(column):
    try:
        return column.type.python_type
    except NotImplementedError:
        if hasattr(column.type, 'impl') and hasattr(column.type.impl, 'python_type'):
            return column.type.impl.python_type
        else:
            raise NotImplementedError(f"The primary key column {pk.name} uses a custom type without a defined `python_type` or suitable `impl` fallback.")
            # this could just warn and return the object as well if it's not that necessary:
            # logging.warning(f"Column {column.name} lacks a python_type and a suitable impl fallback.")
            # return object


def _get_primary_keys(model: Type[DeclarativeBase]) -> Sequence[Column]:
    inspector = inspect(model).mapper
    primary_key_columns = inspector.primary_key
    return primary_key_columns
```

And `_get_python_type` (a more explicit name would be good) would be used in the dict comprehension